src/components/ha-sidebar.ts

```diff
@@ -224,6 +226,13 @@ class HaSidebar extends SubscribeMixin(LitElement) {
   })
   private _hiddenPanels: string[] = [];
 
+  @storage({
```

Comment:
I think a context may be more suitable given the ha-sidebar is 3 levels higher than the automation page where we know if it's dirty or not?

Reply:
yeah 👍

Reply:
> I think a context may be more suitable

Do you mind to elaborate a bit? I'm not very familiar with this concept.

Reply:
You can find more about it here: https://lit.dev/docs/data/context/
We already use it to prevent us bringing registries (like the device and entity registry) to be passed through each component. It's a bit the same like storages except it's kept in-memory thus not persistent, allowing to inject it into a components makes the context available. 

The consumer will be the ha-sidebar, the provider should be the automation-editor component where _dirty is available to know if there are unsaved changes, and then lastly you have the context object which should keep the boolean value. 

It's not the easiest thing from lit to get working unfortunately.

Reply:
The consumer should be higher in the tree than the provider I think, as it works with events. Another thing that will work will for this is Signals. Currently a proposal with ponyfill: <https://github.com/tc39/proposal-signals>

Reply:
> The consumer should be higher in the tree than the provider I think, as it works with events.

Are you saying sidebar cannot be the context consumer, as it is not strictly a parent element of the thing providing the context?

Tried adding `@consume` in ha-sidebar but never got it to pick up anything.

Reply:
Yeah, thats what I'm saying

Reply:
I did try doing this with events, and got it partially working, but I seem to have got stuck on one aspect. 

I can send an event from automation page whether or not the page is dirty or clean, and that is working, but if I'm on the automation page and it is dirty, and I press the browser back button, then what seems to happen is the browser leaves the automation page, but the sidebar still thinks something is dirty so it is still showing the confirm dialog even though we're not on a dirty page anymore. 

In the current proposal with `@storage` I can clear the dirty stored flag on `disconnectedCallback` when we're leaving the automation page for any reason. If I try firing an event to clear dirty from `disconnectedCallback`, it doesn't seem to reach anywhere, presumably because it's no longer connected anymore. So I don't know how to reset the dirty state when we're leaving a dirty page via browser back. (I can't show a confirm dialog on back button either, but I'm ignoring that for now)